### PR TITLE
feat(incidents): Add IncidentSeen model

### PR DESCRIPTION
**Pull request by lynnagara**
Similar to GroupSeen, we'll use this to keep track of users who
have viewed a specific incident.

Ref: SEN-592

### CODE REVIEW

File: src/sentry/incidents/models.py [hunk 1]
```diff
@@ -41,6 +42,19 @@ class Meta:
         unique_together = (('group', 'incident'), )
 
 
+class IncidentSeen(Model):
+    __core__ = False
+
+    incident = FlexibleForeignKey('sentry.Incident')
+    user = FlexibleForeignKey(settings.AUTH_USER_MODEL, db_index=False)
```
Comment (wedamija) [inline src/sentry/incidents/models.py]: I'm not totally sure how safe it is to make a foreign key to `auth_user`. Any thoughts @mattrobenolt? Could be worth marking the migration as `is_dangerous` and running it manually just to be sure.

**Review by wedamija — COMMENTED** [review]
Looks good, just want confirmation on the db stuff from matt. Could you link to this issue: https://getsentry.atlassian.net/browse/SEN-592

File: src/sentry/incidents/models.py (hunk 1 above)
Comment (mattrobenolt) [reply src/sentry/incidents/models.py]: I think it's fine, but yeah, ifo you wanna be safe, let's do it offline, then we'll have a better sense of it for the future. I don't have a better answer off the top of my head, but I _think_ it'll be fine.

File: src/sentry/south_migrations/0478_add_model_IncidentSeen.py [hunk 2]
```diff
@@ -0,0 +1,1344 @@
...
+from south.db import db
+from south.v2 import SchemaMigration
+from django.db import models
+
+
+class Migration(SchemaMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = False
```
Comment (wedamija) [inline src/sentry/south_migrations/0478_add_model_IncidentSeen.py]: Let's mark this as `is_dangerous = True`. Then you can just ask #ops to run this migration manually.

You'll need to provide them with the DDL to actually run, you can generate it with:
`sentry django south_migrate sentry 0478 --verbosity=2`

Comment (lynnagara) [reply src/sentry/south_migrations/0478_add_model_IncidentSeen.py]: Cool - posting here for posterity
```
CREATE TABLE "sentry_incidentseen" ("id" bigserial NOT NULL PRIMARY KEY, "incident_id" bigint NOT NULL, "user_id" integer NOT NULL, "last_seen" timestamp with time zone NOT NULL);
ALTER TABLE "sentry_incidentseen" ADD CONSTRAINT "incident_id_refs_id_74273623" FOREIGN KEY ("incident_id") REFERENCES "sentry_incident" ("id") DEFERRABLE INITIALLY DEFERRED;
CREATE INDEX "sentry_incidentseen_incident_id" ON "sentry_incidentseen" ("incident_id");
ALTER TABLE "sentry_incidentseen" ADD CONSTRAINT "user_id_refs_id_d6a00348" FOREIGN KEY ("user_id") REFERENCES "auth_user" ("id") DEFERRABLE INITIALLY DEFERRED;
```

Comment (wedamija) [reply src/sentry/south_migrations/0478_add_model_IncidentSeen.py]: Looks like the `UNIQUE INDEX` is missing there?

Comment (lynnagara) [reply src/sentry/south_migrations/0478_add_model_IncidentSeen.py]: Thank you for spotting - updated

Comment (wedamija) [reply src/sentry/south_migrations/0478_add_model_IncidentSeen.py]: Looks good! You can post in ops and ask them to run that sql, then afterwards we can merge/deploy

**Review by wedamija — APPROVED** [review]

**Pull request merged**